### PR TITLE
SVC-3551: Configure NCSA GitLab via puppet

### PR DESCRIPTION
**Pull request by billglick**
Add firewall rules for http/https
Open http/https to public
Add GitLab backups via cron
Add sshd matchblock for user git
Update pdk version
Disallow LOCAL git access

See https://jira.ncsa.illinois.edu/browse/SVC-3551

This is being tested on `git-test`.
The logic specific to NCSA GitLab (`git` & `git-test`) is in the `asd-pup-control` repo.

### CODE REVIEW

**Review by bsper2 — CHANGES_REQUESTED** [review]
Some minor changes

File: README.md [hunk 1]
```diff
@@ -9,36 +9,42 @@ NCSA Common Puppet Profiles - configure GitLab service
...
 
+
 ## Usage
 
 The goal is that no paramters are required to be set. The default paramters should work for most NCSA deployments out of the box.
 
-## Reference
 
-See: [REFERENCE.md](REFERENCE.md)
+## Dependencies
```
Comment (bsper2) [inline README.md:35]: Does ::gitlab need to be included as a dependency? Unless I missed something it doesn't look like that class is defined in this module.

File: manifests/update_report.txt [hunk 2]
```diff
@@ -0,0 +1,159 @@
+/* Report generated by PDK at 2021-10-28 17:15:13 -0500 */
```
Comment (bsper2) [inline manifests/update_report.txt]: I think this file got put in when you did the pdk update and can be deleted.

**Review by bsper2 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (billglick) [reply README.md:35]: Yes. Thanks.

**Review by bsper2 — APPROVED** [review]
Looks good thanks.

**Pull request merged**